src/System.Private.CoreLib/shared/System/Globalization/CultureInfo.cs

```diff
@@ -1095,169 +1094,42 @@ private void VerifyWritable()
         internal bool HasInvariantCultureName => Name == InvariantCulture.Name;
 
         /// <summary>
-        /// Helper function overloads of GetCachedReadOnlyCulture.  If lcid is 0, we use the name.
-        /// If lcid is -1, use the altName and create one of those special SQL cultures.
+        /// Gets a cached copy of the specified culture from an internal
+        /// hashtable (or creates it if not found). (LCID version)
         /// </summary>
-        internal static CultureInfo? GetCultureInfoHelper(int lcid, string? name, string? altName)
+        public static CultureInfo GetCultureInfo(int culture)
         {
-            // retval is our return value.
-            CultureInfo? retval;
-
-            // Temporary hashtable for the names.
-            Dictionary<string, CultureInfo>? tempNameHT = s_NameCachedCultures;
-
-            if (name != null)
+            if (culture <= 0)
             {
-                name = CultureData.AnsiToLower(name);
+                throw new ArgumentOutOfRangeException(nameof(culture), SR.ArgumentOutOfRange_NeedPosNum);
             }
 
-            if (altName != null)
-            {
-                altName = CultureData.AnsiToLower(altName);
-            }
+            Dictionary<int, CultureInfo> lcidTable = CachedCulturesByLcid;
+            CultureInfo? result;
 
-            // We expect the same result for both hashtables, but will test individually for added safety.
-            if (tempNameHT == null)
-            {
-                tempNameHT = new Dictionary<string, CultureInfo>();
-            }
-            else
+            lock (lcidTable)
```

Comment:
>lock (lcidTable) [](start = 12, length = 16)

Is there a way try avoid the locking.

Reply:
> Is there a way try avoid the locking

Using a `ConcurrentDictionary<>` would enable lock-free reads (the dictionary would internally still take a lock on writes), but `ConcurrentDictionary<>` isn't currently available in corelib.

A `Hashtable` would enable lock-free reads as well, but you wouldn't want to use it for the LCID case, as the key there is an `Int32`, and thus every read would end up boxing.  It could be used for the lookup-by-name case if desired.

I considered doing the Hashtable thing as well, but didn't want to do it as part of this.